### PR TITLE
Dsnpi 112/google analytics

### PR DESCRIPTION
**Pull request by gabimartini**
Ticket: https://tpximpact.atlassian.net/browse/DSNPI-112?atlOrigin=eyJpIjoiYmIzZjkzOTNlY2Q5NGE3OTg0ZmE0YjBhZDkzYzlhMmIiLCJwIjoiaiJ9

For test you can change the enviroment for anything (I just add it to prevent work locally)

### CODE REVIEW

File: src/components/breadcrumbs/index.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Link from "next/link";
 
 type BreadcrumbsInfo = {
-  name: string;
+  name?: string;
```
Comment (duncanheron) [inline src/components/breadcrumbs/index.tsx:4]: is this required as part of this change or something else you noticed?

File: src/components/layout/index.tsx [hunk 2]
```diff
@@ -4,30 +4,57 @@ import CookiesBanner from "@/components/cookies";
...
+  const [isConsentCookie, setIsConsentCookie] = useState(false);
   const [favicon, setFavicon] = useState("");
+  const [ga, setGa] = useState("");
+
+  const enviroment = process.env.NEXT_PUBLIC_ENVIROMENT;
   useEffect(() => {
-    const getLocalStorageCookies = localStorage.getItem("cookies");
-    const globalContent = localStorage.getItem("globalInfo") || "{}";
-    const { favicon } = JSON.parse(globalContent);
+    const getLocalStorageCookies = healpLocalStorage({
```
Comment (duncanheron) [inline src/components/layout/index.tsx:18]: should this be helpLocalStorage?
What is it doing, can we think of a name that better describes what its helping with?

File: src/components/breadcrumbs/index.tsx (hunk 1 above)
Comment (gabimartini) [reply src/components/breadcrumbs/index.tsx:4]: That was a complain that name could be undefined

File: src/components/layout/index.tsx (hunk 2 above)
Comment (gabimartini) [reply src/components/layout/index.tsx:18]: I wanted to call useLocalStorage, but use is a hook and can't be called inside useEffect, I couldn't come with a better name. That is a function for all localStorage, instead to add inline

File: src/components/layout/index.tsx [hunk 3]
```diff
@@ -4,30 +4,57 @@ import CookiesBanner from "@/components/cookies";
...
+import GoogleAnalytics from "../google-analytics";
+import { getLocalStorage } from "../../../util/localStorageHelper";
 
 export default function Layout({ children }: { children: React.ReactNode }) {
   const [isShowCookie, setIsShowCookie] = useState(true);
+  const [isConsentCookie, setIsConsentCookie] = useState(false);
   const [favicon, setFavicon] = useState("");
+  const [ga, setGa] = useState("");
+
+  const enviroment = process.env.NEXT_PUBLIC_ENVIROMENT;
```
Comment (madz-lw) [inline src/components/layout/index.tsx:16]: Can the spelling change to environment instead of enviroment please 😊 In the const variable and the .env

Comment (gabimartini) [reply src/components/layout/index.tsx]: Thanks Maddie, it's done :)

**Review by madz-lw — APPROVED** [review]
LGTM thanks Gabi 😊

**Pull request merged**